vault/token_store.go

```diff
@@ -613,11 +741,20 @@ func (ts *TokenStore) handleCreateCommon(
 		te.ID = data.ID
 	}
 
-	// Only permit policies to be a subset unless the client has root or sudo privileges
-	if len(data.Policies) == 0 {
+	switch {
+	// If we have a role, we don't even consider parent policies; the role
+	// allowed policies trumps all
+	case role != nil:
+		if !strListSubset(role.AllowedPolicies, data.Policies) {
```

Comment:
What happens if data specified no policies? Should the role based still inherit the allowed policies? Otherwise it gets the empty list.